sdk/core/azure-core/src/main/java/com/azure/core/util/metrics/Meter.java

```diff
@@ -0,0 +1,194 @@
+     *
+     * &#47;&#47; on connection initialized:
+     * activeHttpConnections.add&#40;1, attributes, currentContext&#41;;
+     *
+     * &#47;&#47; on connection closed:
+     * activeHttpConnections.add&#40;-1, attributes, currentContext&#41;;
+     * </pre>
+     * <!-- end com.azure.core.util.metrics.Meter.upDownCounter -->
+     *
+     * @param name short counter  name following https://opentelemetry.io/docs/reference/specification/metrics/api/#instrument-naming-rule
+     * @param description free-form text describing the counter
+     * @param unit optional unit of measurement.
+     * @return new instance of {@link LongCounter}
+     * @throws NullPointerException if name or description is null.
+     */
+    LongCounter createLongUpDownCounter(String name, String description, String unit);
+
+    /**
+     * Creates and returns attribute collection implementation specific to the meter implementation.
+     * Attribute collections differ in how they support different types of attributes and internal
+     * data structures they use.
+     *
+     * For the best performance, client libraries should create and cache attribute collections
+     * for the client lifetime and pass cached instance when recoding new measurements.
+     *
+     * <!-- src_embed com.azure.core.util.metrics.Meter.longCounter#errorFlag -->
+     * <pre>
+     *
+     * &#47;&#47; Create attributes for possible error codes. Can be done lazily once specific error code is received.
+     * TelemetryAttributes successAttributes = defaultMeter.createAttributes&#40;new HashMap&lt;String, Object&gt;&#40;&#41; &#123;&#123;
+     *         put&#40;&quot;endpoint&quot;, &quot;http:&#47;&#47;service-endpoint.azure.com&quot;&#41;;
+     *         put&#40;&quot;error&quot;, true&#41;;
+     *     &#125;&#125;&#41;;
+     *
+     * TelemetryAttributes errorAttributes =  defaultMeter.createAttributes&#40;new HashMap&lt;String, Object&gt;&#40;&#41; &#123;&#123;
+     *         put&#40;&quot;endpoint&quot;, &quot;http:&#47;&#47;service-endpoint.azure.com&quot;&#41;;
+     *         put&#40;&quot;error&quot;, false&#41;;
+     *     &#125;&#125;&#41;;
+     *
+     * LongCounter httpConnections = defaultMeter.createLongCounter&#40;&quot;az.core.http.connections&quot;,
+     *     &quot;Number of created HTTP connections&quot;, null&#41;;
+     *
+     * boolean success = false;
+     * try &#123;
+     *     success = doThings&#40;&#41;;
+     * &#125; finally &#123;
+     *     httpConnections.add&#40;1, success ? successAttributes : errorAttributes, currentContext&#41;;
+     * &#125;
+     *
+     * </pre>
+     * <!-- end com.azure.core.util.metrics.Meter.longCounter#errorFlag -->
+     * @param attributeMap map of key value pairs to cache.
+     * @return an instance of {@code AttributesBuilder}
+     */
+    TelemetryAttributes createAttributes(Map<String, Object> attributeMap);
+
+    /**
+     * Closes the token manager.
+     */
+    @Override
+    void close();
```

Comment:
This can be removed.

Reply:
it removes the exception clause from the signature, I'd prefer to keep it, and force meter implementations handle exceptions.